vault/templates/role-vault.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .Chart.Name }}-init-permissions
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "-1"
+    "helm.sh/hook-delete-policy": before-hook-creation, hook-succeeded
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["list"]
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["create"]
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["update"]
```

Comment:
you can combine verbs into a single array

Reply:
I'm using separate array for 'update' because it's limited to corresponding resource 'vault-init-secrets'.
```yaml
  - apiGroups: [""]
    resources: ["secrets"]
    verbs: ["create", "update"]
    resourceNames: ["vault-init-secrets"]
```
If i use combined array for this action we'll get error on creating secret:
```
"secrets is forbidden: User \"system:serviceaccount:vault-system:auth-vault-init\" cannot create resource \"secrets\" in API group \"\" in the namespace \"vault-system\""
```